src/Eloquent/SoftDeletes.php

```diff
@@ -4,6 +4,7 @@
 
 namespace MongoDB\Laravel\Eloquent;
 
+/** @deprecated 6.0.0 in favor of \Illuminate\Database\Eloquent\SoftDeletes */
```

Comment:
We can trigger a deprecation when the PHP file is loaded.

In the `@deprecated` annotation, we usually give the version of the deprecation. 

```suggestion
trigger_error(sprintf('Since mongodb/laravel-mongodb:5.5, trait "%s" is deprecated, use "%s" instead.', SoftDeletes::class, \Illuminate\Database\Eloquent\SoftDeletes::class), E_USER_DEPRECATED);

/** @deprecated since mongodb/laravel-mongodb:5.5, use \Illuminate\Database\Eloquent\SoftDeletes instead */
```

Example of an other deprecation I put in the library: https://github.com/mongodb/laravel-mongodb/blob/b02b40c120be6dee9ebf63ff0b7bc42464c8fce4/src/Connection.php#L126-L132

An other example of deprecating a class in Symfony: https://github.com/symfony/symfony/blob/138109a63e2202674d81f31481da7c6defe4c0ce/src/Symfony/Component/Serializer/CacheWarmer/CompiledClassMetadataCacheWarmer.php#L19